base/linalg/generic.jl

```diff
@@ -550,6 +574,21 @@ dot(x::Number, y::Number) = vecdot(x, y)
     ⋅(x,y)
 
 Compute the dot product. For complex vectors, the first vector is conjugated.
+
+```jldoctest
+julia> a = [1; 1]
+2-element Array{Int64,1}:
+ 1
+ 1
+
+julia> b = [2; 3]
+2-element Array{Int64,1}:
+ 2
+ 3
+
+julia> dot(a, b)
+5
+```
```

Comment:
Perhaps an example of conjugation in the complex case would be useful, e.g. `dot([im, im], [1, 1])`?